tests/utils/testapp/gen-syscall-events-callstack/gen-syscall-events-callstack.c

```diff
@@ -40,13 +40,25 @@ long __attribute__ ((noinline))
 my_gettid(void)
 {
     long ret;
+#ifdef __x86_64
     asm volatile
     (
         "syscall"
         : "=a" (ret)
         : "0"(__NR_gettid)
         : "cc", "rcx", "r11", "memory"
     );
+#elif __i386
+    asm volatile
+    (
+        "int $0x80"
+        : "=a" (ret)
+        : "0"(__NR_gettid)
+        : "cc", "edi", "esi", "memory"
+    );
+#else
+#error "Userspace callstack test not supported for this architecture."
```

Comment:
This will fail the build on all non-x86/x64 platforms. We should remove that test app from the build on those platforms.
That can be achieved by removing the folder from the in the parent `Makefile.am`as done here https://github.com/lttng/lttng-tools/blob/master/tests/regression/Makefile.am#L34.

I'm not sure what's the best way to detect the target architecture from autoconf and export that... I'm looking into it.

Reply:
You're totally right. Having a #error here is a bad idea. We should simply prevent this binary from being compiled on unsupported architectures.

Reply:
#error when arch is unsupported is fine. But it should come along with arch detection in makefiles. lttng-ust and liburcu both do it.